src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/FixAllContextHelper.cs

```diff
@@ -50,6 +54,48 @@ internal static class FixAllContextHelper
 
                     break;
 
+                case FixAllScope.ContainingMember or FixAllScope.ContainingType:
+                    if (document != null && !await document.IsGeneratedCodeAsync(cancellationToken).ConfigureAwait(false))
+                    {
+                        if (!fixAllSpans.IsEmpty)
+                        {
+                            // We are processing a span-based FixAllContext created for a partial declaration.
+                            var documentDiagnostics = ImmutableArray<Diagnostic>.Empty;
+                            foreach (var fixAllSpan in fixAllSpans)
+                            {
+                                documentDiagnostics = documentDiagnostics.AddRange(await fixAllContext.GetDocumentSpanDiagnosticsAsync(
+                                    document, fixAllSpan).ConfigureAwait(false));
```

Comment:
can this helper just take in many spans?  woudl that be more efficient down below instead of calling multiple times into it?

Reply:
As explained in https://github.com/dotnet/roslyn/pull/59989#discussion_r821315635, multiple fix all spans is going to be an extreme corner case here that we do not need to optimize for. For all practical purposes and performance measurements, we should just assume a single FixAll span within a document.